File: node-api-gateway/internal/server/interface/http/controller/controller.js
class Controller {
  constructor() {

  }
}

Controller.prototype.success = function(res) {
  const inner = function(status=200, message="Success", data=null) {
    return res.status(parseInt(status)).send({
      status: status,
      message: message,
      data: data
    })
  }
  return inner;
}

Controller.prototype.failure = function(res) {
  const inner = function(status=404, message="Error", data=null) {
    return res.status(parseInt(status)).send({
      status: status,
      message: message,
      data: data
    })
  }
  return inner;
}


module.exports.Controller = Controller;

File: rtsp_sender/apps/rtsp_sender/interface/http_interface/controller/controller.go
package controller

import (
	"github.com/dathuynh1108/hcmut-thexis/rtsp-sender/apps/rtsp_sender/entity"
	"github.com/gofiber/fiber/v2"
)

type controller struct {
}

func (c *controller) Success(ctx *fiber.Ctx, successCode int, code int, message string, data interface{}) error {
	response := entity.Response{
		Code:    code,
		Message: message,
		Data:    data,
	}
	return ctx.Status(successCode).JSON(response)
}
func (c *controller) Failure(ctx *fiber.Ctx, errorCode int, code int, message string) error {
	response := entity.Response{
		Code:    code,
		Message: message,
		Data:    nil,
	}
	return ctx.Status(errorCode).JSON(response)
}
